File: src/main/java/block/BlockLeaves.java
package block;

import gl.UVPair;
import registry.RegistryName;

public class BlockLeaves extends Block {
    private LogType _type;

    public BlockLeaves(LogType type) {
        super(new RegistryName("blockgame", type.getTypeName()+"_leaves"));
        _type = type;
    }

    @Override
    public boolean isOpaque() {
        return false;
    }

    @Override
    public EnumRenderLayer renderLayer() {
        return EnumRenderLayer.TRANSLUCENT;
    }

    @Override
    public UVPair getUVForFace(EnumDirection face) {
        return _type.getLeaves();
    }
}

File: src/main/kotlin/block/BlockLeaves.kt
package block

import gl.UVPair
import registry.RegistryName

class BlockLeaves(val type: LogType) : Block( RegistryName("blockgame", "${type.name}_leaves")) {
    override fun isOpaque(): Boolean {
        return false
    }

    override fun getUVForFace(face: EnumDirection): UVPair {
        return type.leaves
    }
}